pkg/providers/instance/types.go

```diff
@@ -0,0 +1,62 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instance
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/samber/lo"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+)
+
+// Instance is an internal data representation of either an ec2.Instance or an ec2.FleetInstance
+// It contains all the common data that is needed to inject into the Machine from either of these responses
+type Instance struct {
```

Comment:
Alternatively, just use ec2.Instance as your data object everywhere, and write a translation from ec2.CreateFleetInstance -> ec2.Instance

Reply:
I did that originally, but @bwagner5 had some concern that this would confuse anyone who expected data fields to be there that wouldn't actually be there. I think representing the data this way is a nice compromise